stl/inc/vector

```diff
@@ -2940,6 +2940,109 @@ _CONSTEXPR20 void _Fill_vbool(_FwdIt _First, _FwdIt _Last, const _Ty& _Val) {
         *_VbFirst                  = (*_VbFirst & _LastDestMask) | (_FillVal & _LastSourceMask);
     }
 }
+
+template <class _InIt, class _Ty>
+_NODISCARD _CONSTEXPR20 _InIt _Find_vbool(_InIt _First, const _InIt _Last, const _Ty& _Val) {
+#ifdef __cpp_lib_bitops // TRANSITION, VSO-1020212
+    // Find _Val in [_First, _Last)
+    if (_First == _Last) {
+        return _First;
+    }
+
+    _Vbase* _VbFirst            = const_cast<_Vbase*>(_First._Myptr);
+    const _Vbase* const _VbLast = _Last._Myptr;
+
+    const auto _FirstSourceMask = static_cast<_Vbase>(-1) << _First._Myoff;

```

Comment:
what's this actually doing? it's unclear if this is a right shift in disguise or what? Why do we need to use -1 here?

Reply:
This is creating a mask, that zeros out the first `_First._Myoff` bits.

`static_cast<_Vbase>(-1) << 2 == [11111111111111111111111111111100]`

This is necessary as we cannot count the previous bits

Reply:
oh I see, perhaps numeric_limits<_Vbase>::max() << 2 would be clearer, or a comment to that effect. I don't know why I thought it was a left shift, it obviously isn't, but I guess on quick reading I swapped the left and right side of the <<.

Reply:
Hm I would keep it as is, because it is both existing practise and all the bit fiddling is nontrivial so to be consitent I would need to massivly comment everything

Reply:
yeah that's fine.